dev/pyRevitLoader/pyRevitExtensionParser/PyRevitConfig.cs

```diff
@@ -0,0 +1,441 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace pyRevitExtensionParser
+{
+    /// <summary>
+    /// Provides access to pyRevit configuration settings stored in an INI file.
+    /// Handles core settings like user extensions, locale, loader options, and extension-specific configurations.
+    /// </summary>
+    public class PyRevitConfig
+    {
+        /// <summary>
+        /// The underlying INI file handler for reading and writing configuration values.
+        /// </summary>
+        private readonly IniFile _ini;
+        
+        /// <summary>
+        /// Cached values for boolean conversion to avoid repeated string allocations.
+        /// </summary>
+        private const string TrueString = "true";
+        private const string FalseString = "false";
+        
+        /// <summary>
+        /// Gets the full path to the configuration file.
+        /// </summary>
+        public string ConfigPath { get; }
+
+        /// <summary>
+        /// Gets or sets the user extensions configuration as a raw string.
+        /// This can be a single path or a Python-style list of paths.
+        /// Use <see cref="UserExtensionsList"/> for parsed list access.
+        /// </summary>
+        /// <remarks>
+        /// Returns null if the value is not set or empty.
+        /// Trailing/leading whitespace is automatically trimmed on read.
+        /// </remarks>
+        public string UserExtensions 
+        {
+            get
+            {
+                var value = _ini.IniReadValue("core", "userextensions");
+                // Avoid Trim() allocation if value is already null/empty
```

Comment:
Can we move these kind of comments in a summary of some sort instead of repeating them throughout?
```suggestion
```